app/assets/javascripts/hyrax/app.js

```diff
@@ -40,6 +41,14 @@ Hyrax = {
       var controls = new CollectionControls($('#collection-controls'));
     },
 
+    // Collection types
+    collectionTypes: function() {
+      var element = $('.collection-types-wrapper')
+      if (element.length > 0) {
+        var CollectionTypes = require('hyrax/collection_types');
```

Comment:
@elrayle I don't have recommendations. Just an observation. I defer to @adamjarling's JS experience.